Simplified/NYPLReadiumViewSyncManager.m

```diff
@@ -205,11 +145,72 @@ - (void)syncReadingPositionForBook:(NSString *)bookID
         [currentLocationString isEqualToString:serverLocationString] ||
         !serverLocationString) {
       weakSelf.shouldPostLastRead = YES;
-    } else {
-      [[NSOperationQueue mainQueue] addOperationWithBlock:^{
-        [[NYPLRootTabBarController sharedController] safelyPresentViewController:alertController animated:YES completion:nil];
-      }];
+      return;
     }
+      
+    [[NSOperationQueue mainQueue] addOperationWithBlock:^{
+      NSDictionary *spineItemDetails = weakSelf.bookMapDictionary[responseJSON[@"idref"]];
```

Comment:
can we move all the code inside the `addOperationWithBlock` closure to a new method? not sure if that will involve a gazillion parameters but it would be nice to break up this long deeply-nested method.